x-pack/plugins/secops/public/pages/hosts/index.tsx

```diff
@@ -34,36 +37,53 @@ interface Props {
 
 export const Hosts = connect()(
   pure<Props>(({ dispatch }) => (
-    <EventsQuery sourceId="default" startDate={startDate} endDate={endDate}>
-      {({ events, kpiEventType, loading }) => (
-        <Pane1FlexContent data-test-subj="pane1FlexContent">
-          <VisualizationPlaceholder>
-            <HorizontalBarChart
-              loading={loading}
-              title="KPI event types"
-              width={490}
-              height={279}
-              barChartdata={
-                kpiEventType.map((i: KpiItem) => ({
-                  x: i.count,
-                  y: i.value,
-                })) as HorizontalBarChartData[]
-              }
-            />
-          </VisualizationPlaceholder>
-          <VisualizationPlaceholder>
-            <BasicTable
-              columns={getEventsColumns(dispatch)}
-              loading={loading}
-              pageOfItems={events}
-              sortField="host.hostname"
-              title="Events"
-            />
-          </VisualizationPlaceholder>
-          <Placeholders timelineId="pane2-timeline" count={8} myRoute="Hosts" />
-        </Pane1FlexContent>
-      )}
-    </EventsQuery>
+    <WithSource sourceId="default">
+      {({ auditbeatIndicesExist }) =>
+        auditbeatIndicesExist || auditbeatIndicesExist === null ? (
+          <EventsQuery sourceId="default" startDate={startDate} endDate={endDate}>
+            {({ events, kpiEventType, loading }) => (
+              <Pane1FlexContent data-test-subj="pane1FlexContent">
+                <VisualizationPlaceholder>
+                  <HorizontalBarChart
+                    loading={loading}
+                    title="KPI event types"
+                    width={490}
+                    height={279}
+                    barChartdata={
+                      kpiEventType.map((i: KpiItem) => ({
+                        x: i.count,
+                        y: i.value,
+                      })) as HorizontalBarChartData[]
+                    }
+                  />
+                </VisualizationPlaceholder>
+                <VisualizationPlaceholder>
+                  <BasicTable
+                    columns={getEventsColumns(dispatch)}
+                    loading={loading}
+                    pageOfItems={events}
+                    sortField="host.hostname"
+                    title="Events"
+                  />
+                </VisualizationPlaceholder>
+                <Placeholders timelineId="pane2-timeline" count={8} myRoute="Hosts" />
+              </Pane1FlexContent>
+            )}
+          </EventsQuery>
+        ) : (
+          <WithKibanaChrome>
+            {({ basePath }) => (
+              <EmptyPage
+                title="Looks like you don't have any auditbeat indices."
+                message="Let's add some!"
+                actionLabel="Setup Instructions"
+                actionUrl={`${basePath}/app/kibana#/home/tutorial_directory/security`}
```

Comment:
Can't you remove 1 level of indentation and the `WithKibanaChrome` by just importing it and using it:

```ts
import chrome from 'ui/chrome';
const basePath = chrome.getBasePath();
... 
actionUrl={`${basePath}/app/kibana#/home/tutorial_directory/security`}
```

It would be one less react class with less unit tests if you decided you would have to write unit tests for the `WithKibanaChrome`

Reply:
Good Idea!!!